corehq/apps/hqwebapp/views.py

```diff
@@ -364,8 +365,13 @@ def _login(req, domain_name, template_name):
 
     context = {}
     custom_landing_page = settings.CUSTOM_LANDING_TEMPLATE
-    if custom_landing_page is not None:
-        template_name = custom_landing_page
+    if custom_landing_page:
```

Comment:
I can see you're preserving current behaviour by not changing the value of `template_name` that was passed into the function. But the view that allows a non-default value to be set is [`domain_login()`][1] and it is only called in [urls.py here][2], where the template_name is left as default (**unless I'm missing something important**).

If you drop the unused(**‽**) `template_name` param in `login()`, `domain_login()` and`_login()`, and don't bother with setting a default in `CUSTOM_LANDING_TEMPLATE`, then determining the value of `template_name` is simpler:
```python
    custom_landing_page = settings.CUSTOM_LANDING_TEMPLATE
    host = req.get_host()
    if isinstance(custom_landing_page, six.string_types):
        template_name = custom_landing_page
    elif host in custom_landing_page:
        template_name = custom_landing_page[host]
    else:
        template_name = 'login_and_password/login.html'
    if domain_name:
        # ...
```
This also allows a custom landing page AND setting `domain_name` to something other than `None` (for enterprise customers with more than one domain), which the current behaviour does not allow. But maybe checking if we want to allow that is out of scope of this PR.

What do you think?


  [1]: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqwebapp/views.py#L416
  [2]: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqwebapp/urls.py#L65

Reply:
I removed this unused parameter https://github.com/dimagi/commcare-hq/commit/a53680499faaf80fc56f8c93c07bbb7e92d75af8 but I'm not going to go through with the full change you suggested because that code is written would set the default to our default commcare template which would be undesirable for enterprise envs in a weird case where a proxy change caused the variables set by `get_host()` to be set differently. For ICDS I plan to set it to:

```python
CUSTOM_LANDING_TEMPLATE = {
   "reach.commcarehq.org": "login_and_password/login.html", # likely something more custom later on
   "default": "icds/login.html",
}
```

This prioritizes enterprise server's main sites and ensures that their main sites would never get sent the wrong default landing page

Reply:
Not to say that this is a great workflow or will last forever, but we've decided that this sort of work is better than maintaining a separate environment on another hosting provider for the next year or longer.

Also I plan on removing the acceptance of string once this work is done and write a changelog for commcare-cloud (echis uses these variables as well, so they will need to be warned about any breaking changes)

Reply:
:+1: That makes sense.